module-services/service-db/ServiceDBCommon.cpp

```diff
@@ -35,10 +35,18 @@ sys::MessagePointer ServiceDBCommon::DataReceivedHandler(sys::DataMessage *msgl,
         auto query     = msg->getQuery();
         auto queryType = query->type;
         auto result    = interface->runQuery(std::move(query));
-        auto recordID  = result->getRecordID().value_or(0);
-        responseMsg    = std::make_shared<db::QueryResponse>(std::move(result));
+        std::optional<uint32_t> val;
+        if (result != nullptr) {
+            val = result->getRecordID();
+        }
+        else {
+            LOG_WARN("There is no response associated with query: %s!", query ? query->debugInfo().c_str() : "");
+        }
+        responseMsg = std::make_shared<db::QueryResponse>(std::move(result));
 
-        sendUpdateNotification(msg->getInterface(), queryType, recordID);
+        if (val) {
+            sendUpdateNotification(msg->getInterface(), queryType, val.value());
```

Comment:
Calling `val.value()` looks a bit strange to me. How about replacing it with `*val` or changing the name of the `val` variable to something more adequate?

Reply:
That's true. I'll rename it?